include/beast/http/string_body.hpp

```diff
@@ -117,14 +120,14 @@ struct string_body
             {
                 body_.resize(len + n);
             }
-            catch(std::length_error const&)
+            catch(std::exception const&)
             {
                 ec = error::buffer_overflow;
                 return;
             }
-            ec.assign(0, ec.category());
             buffer_copy(boost::asio::buffer(
                 &body_[0] + len, n), buffers);
```

Comment:
I would always call append.
The container should know how to handle allocation and copying at best...
Better than we know...
So resize() would be obsolete too. But people told me, that resize() is VERY important.

Or do you mean an optimization for 1 character only?! Like push_back() ?